src/gss_names.c

```diff
@@ -626,9 +708,104 @@ uint32_t gssntlm_inquire_name(uint32_t *minor_status,
+    }
+    *attrs = GSS_C_NO_BUFFER_SET;
+
+    if (in->attrs == NULL) {
+        return GSSERRS(0, GSS_S_COMPLETE);
+    }
+
+    for (size_t i = 0; in->attrs[i].attr_name != NULL; i++) {
+        struct gssntlm_name_attribute *attr = &in->attrs[i];
+        size_t attr_name_len = strlen(attr->attr_name);
+
+        gss_buffer_t attr_value = &attr->attr_value;
+        /* +1 for '=' separator and +1 for EOL */
+        buf.length = attr_value->length + attr_name_len + 2;
+        buf.value = malloc(buf.length);
+        if (buf.value == NULL) {
+            retmin = ENOMEM;
+            goto done;
+        }
+
+        /* Construct 'attr_name=<attr_value>\0' string */
+        memcpy(buf.value, attr->attr_name, attr_name_len);
+        ((char*)buf.value)[attr_name_len] = '=';
+        memcpy(((char*)buf.value) + attr_name_len + 1, attr_value->value, attr_value->length);
+        ((char*)buf.value)[buf.length] = 0;
+
+        retmaj = gss_add_buffer_set_member(&retmin, &buf, attrs);
+        free(buf.value);
+        if (retmaj != GSS_S_COMPLETE) goto done;
+    }
+    return GSSERRS(0, GSS_S_COMPLETE);
+
+done:
+    (void)gss_release_buffer_set(&tmpmin, attrs);
+    return GSSERRS(retmin, retmaj);
+}
+
+/* RFC6680 - GSSAPI Naming Extensions */
+uint32_t gssntlm_get_name_attribute(uint32_t *minor_status,
+                                    gss_name_t name,
+                                    gss_buffer_t attr,
+                                    int *authenticated,
+                                    int *complete,
+                                    gss_buffer_t value,
+                                    gss_buffer_t display_value,
+                                    int *more)
+{
+    uint32_t retmin;
+    uint32_t retmaj;
+    const struct gssntlm_name *in = (const struct gssntlm_name *)name;
+    struct gssntlm_name_attribute *found_attr;
+
+    if (name == GSS_C_NO_NAME) {
+        return GSSERRS(GSS_S_BAD_NAME, GSS_S_CALL_INACCESSIBLE_READ);
+    }
+    if (attr == NULL) {
+        return GSSERRS(ERR_NOARG, GSS_S_CALL_INACCESSIBLE_READ);
+    }
+
+    if (display_value != NULL) {
+        display_value->value = NULL;
+        display_value->length = 0;
+    }
+    if (more != NULL) *more = 0;
+    if (authenticated) *authenticated = 0;
+    if (complete) *complete = 0;
+
+    found_attr = gssntlm_find_attr(in->attrs, attr->value, attr->length);
+    if (!found_attr) {
+        return GSSERRS(GSS_S_UNAVAILABLE, ENOENT);
+    }
+
+    if (authenticated) *authenticated = 1;
+    if (complete) *complete = 1;
+    if (value != NULL) {
+        gss_buffer_t attr_value = &found_attr->attr_value;
+        value->value = malloc(attr_value->length);
+        if (value->value == NULL) {
+            return GSSERRS(GSS_S_FAILURE, ENOMEM);
+        }
+        memcpy(value->value, attr_value->value, attr_value->length);
+        value->length = attr_value->length;
+    }
+    return GSSERRS(0, GSS_S_COMPLETE);
 }
 
+
```

Comment:
Extra whitespace.

Reply:
Oops, how can it be? I have wasted the whole line of visible space on your screen...
Sorry, just kidding :-) Frankly, I really think that times of 80x25 characters on the screen are gone forever so you can weaken the rules. Will fix it...

Reply:
I use 5 vim buffers side by side on my screen to cross reference code, and 80 columns maintains it readable, and in general avoids me having to scroll my head sideways within the specific code section. I can care less about 80x25 screens.

Reply:
LOL!
But you need to consider scroll your head at least sometimes to prevent a neckache :-)

Once I have a rule how to resolve 'it doesn't fit to 80 characters' issue, I don't mind.
But what if that's just not possible? I have seen a C++ code with class name of 120+ characters,
and having definition of constructor in form ClassName::ClassName() this length is doubled in a single line.
So what will you in such a case?

Reply:
One of the many reasons I dislike C++ :-D
And I do not have C++ projects and for projects that do not care I will simply be less efficient and do what their style wants. It's like people that send HTML emails, nothing much I can do about that, I just tend to ignore them unless I have to deal with them :-D